dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/appmanager/JCloudsAppStorageService.java

```diff
@@ -271,6 +271,63 @@ public void cleanUp()
         return reservedNamespaces;
     }
 
+    private boolean validateApp( App app, Cache<App> appCache )
```

Comment:
A unit test would be appropriate here, as this code can easily be tested. Method could be made public (or just default modifier). Can wait until after hard freeze.